wallet/src/main/scala/org/bitcoins/wallet/models/AddressTagTable.scala

```diff
@@ -0,0 +1,55 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.wallet.utxo.{AddressTag, CustomAddressTag}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+
+case class AddressTagDb(
+    address: BitcoinAddress,
+    tag: AddressTag,
+    tagType: String) {
+  require(
+    tag.tagType.isEmpty || tag.tagType == tagType,
```

Comment:
So I don't really understand why we have `tagType` as a parameter. Why can't we just call `tag.tagType` to get the type and remove the `tagType` parameter all together?

Reply:
Because sometimes (like in the db mappers) we don't have access to the tag type when we are getting the tag, so we need to be able to pass it in.